include/zephyr/posix/unistd.h

```diff
@@ -55,6 +55,153 @@ pid_t getpid(void);
+#define __z_posix_sysconf_SC_SYMLOOP_MAX		  (-1L)
+#define __z_posix_sysconf_SC_TIMER_MAX			  CONFIG_MAX_TIMER_COUNT
+#define __z_posix_sysconf_SC_TTY_NAME_MAX		  (-1L)
+#define __z_posix_sysconf_SC_TZNAME_MAX			  (-1L)
+#define __z_posix_sysconf_SC_ADVISORY_INFO		  (-1L)
+#define __z_posix_sysconf_SC_BARRIERS			  COND_CODE_1(CONFIG_PTHREAD_IPC,	  \
+								      (SUPPORTED_POSIX1_VERSION), \
+								      (-1L))
+#define __z_posix_sysconf_SC_ASYNCHRONOUS_IO		  (-1L)
+#define __z_posix_sysconf_SC_CLOCK_SELECTION		  (-1L)
+#define __z_posix_sysconf_SC_CPUTIME			  (-1L)
+#define __z_posix_sysconf_SC_FSYNC			  (-1L)
+#define __z_posix_sysconf_SC_IPV6			  (-1L)
+#define __z_posix_sysconf_SC_JOB_CONTRO			  (-1L)
+#define __z_posix_sysconf_SC_MAPPED_FILES		  (-1L)
+#define __z_posix_sysconf_SC_MEMLOCK			  (-1L)
+#define __z_posix_sysconf_SC_MEMLOCK_RANGE		  (-1L)
+#define __z_posix_sysconf_SC_MEMORY_PROTECTION		  (-1L)
+#define __z_posix_sysconf_SC_MESSAGE_PASSING		  (-1L)
+#define __z_posix_sysconf_SC_MONOTONIC_CLOCK		  (-1L)
+#define __z_posix_sysconf_SC_PRIORITIZED_IO		  (-1L)
+#define __z_posix_sysconf_SC_PRIORITY_SCHEDULING	  (-1L)
+#define __z_posix_sysconf_SC_RAW_SOCKETS		  (-1L)
+#define __z_posix_sysconf_SC_READER_WRITER_LOCKS	  (-1L)
+#define __z_posix_sysconf_SC_REALTIME_SIGNALS		  (-1L)
+#define __z_posix_sysconf_SC_REGEXP			  (-1L)
+#define __z_posix_sysconf_SC_SAVED_IDS			  (-1L)
+#define __z_posix_sysconf_SC_SEMAPHORES			  COND_CODE_1(CONFIG_PTHREAD_IPC,	  \
+								      (SUPPORTED_POSIX1_VERSION), \
+								      (-1L))
+#define __z_posix_sysconf_SC_SHARED_MEMORY_OBJECTS	  (-1L)
+#define __z_posix_sysconf_SC_SHELL			  (-1L)
+#define __z_posix_sysconf_SC_SPAWN			  (-1L)
+#define __z_posix_sysconf_SC_SPIN_LOCKS			  (-1L)
+#define __z_posix_sysconf_SC_SPORADIC_SERVER		  (-1L)
+#define __z_posix_sysconf_SC_SS_REPL_MAX		  (-1L)
+#define __z_posix_sysconf_SC_SYNCHRONIZED_IO		  (-1L)
+#define __z_posix_sysconf_SC_THREAD_ATTR_STACKADDR	  COND_CODE_1(CONFIG_PTHREAD_IPC,	  \
+								      (SUPPORTED_POSIX1_VERSION), \
+								      (-1L))
+#define __z_posix_sysconf_SC_THREAD_ATTR_STACKSIZE	  COND_CODE_1(CONFIG_PTHREAD_IPC,	  \
+								      (SUPPORTED_POSIX1_VERSION), \
+								      (-1L))
+#define __z_posix_sysconf_SC_THREAD_CPUTIME		  (-1L)
+#define __z_posix_sysconf_SC_THREAD_PRIO_INHERIT	  (-1L)
+#define __z_posix_sysconf_SC_THREAD_PRIO_PROTECT	  (-1L)
+#define __z_posix_sysconf_SC_THREAD_PRIORITY_SCHEDULING	  COND_CODE_1(CONFIG_PTHREAD_IPC,	  \
+								      (SUPPORTED_POSIX1_VERSION), \
+								      (-1L))
+#define __z_posix_sysconf_SC_THREAD_PROCESS_SHARED	  (-1L)
+#define __z_posix_sysconf_SC_THREAD_ROBUST_PRIO_INHERIT	  (-1L)
+#define __z_posix_sysconf_SC_THREAD_ROBUST_PRIO_PROTECT	  (-1L)
+#define __z_posix_sysconf_SC_THREAD_SAFE_FUNCTIONS	  (-1L)
+#define __z_posix_sysconf_SC_THREAD_SPORADIC_SERVER	  (-1L)
+#define __z_posix_sysconf_SC_THREADS			  (-1L)
+#define __z_posix_sysconf_SC_TIMEOUTS			  (-1L)
```

Comment:
This should probably be defined to `_POSIX_VERSION` - the spec says 

> Support for timeouts for some blocking services is mandatory in POSIX.1-2017.

but that is not super descriptive of which functions need to support this option. In any case, we make every effort to support timeouts in all POSIX API calls, so it should be a given.